### DISCUSSION TITLE
Как избавиться от Future<void> у accept?

### DISCUSSION THREAD

**Discussion by IlyaMax** [surfstudio/SurfGear#214] [Ideas]
Какие проблемы это вызывает мы уже обсуждали. Я попробовал выпилить его оттуда и не совсем получилось. Дело в том, что при добавлении значения в стрим оно попадает туда не сразу, а в микротаске(см.доку). То-есть не гарантируется последовательность входящих значений. Самый простой способ это решить, который я вижу это использовать синхронный сабжект. Насколько это ок? По-моему мы и не используем стейт как-то по-другому на самом деле, но не точно. Какие есть еще варианты?

**Comment by IlyaMax** [comment]
При чем если брать по value то всё будет синхронно и последовательно. Но при подписке будут проблемы

**Reply by IlyaMax** [reply]
Хотя возможно я и не прав, насчет последовательности. Микротаски может по-другому как-то работают.

**Comment by saturov** [comment]
Удалось ли воспроизвести ситуацию, когда элементы приходят в стрим не в том порядке, в котором акцептятся туда?

**Reply by IlyaMax** [reply]
Нет, не пробовал
